lib/tests/streamlit/web/server/routes_test.py

```diff
@@ -155,6 +157,83 @@ def test_parse_url_path_404(self):
             assert r.code == 404
 
 
+class AppStaticFileHandlerTest(tornado.testing.AsyncHTTPTestCase):
+    def setUp(self) -> None:
+        self._tmpdir = tempfile.TemporaryDirectory(dir=os.getcwd())
+        self._tmpfile = tempfile.NamedTemporaryFile(dir=self._tmpdir.name, delete=False)
+        self._tmp_js_file = tempfile.NamedTemporaryFile(
+            dir=self._tmpdir.name, suffix="script.js", delete=False
+        )
+        self._tmp_image_file = tempfile.NamedTemporaryFile(
+            dir=self._tmpdir.name, suffix="image.png", delete=False
+        )
+
+        self._symlink_outside_directory = "symlink_outside"
+        self._symlink_inside_directory = "symlink_inside"
+
+        os.symlink(
+            "/", os.path.join(self._tmpdir.name, self._symlink_outside_directory)
+        )
+        os.symlink(
+            self._tmpfile.name,
+            os.path.join(self._tmpdir.name, self._symlink_inside_directory),
+        )
+
+        self._filename = os.path.basename(self._tmpfile.name)
+        self._js_filename = os.path.basename(self._tmp_js_file.name)
+        self._image_filename = os.path.basename(self._tmp_image_file.name)
+
+        super().setUp()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self._tmpdir.cleanup()
+
+    def get_app(self):
+        return tornado.web.Application(
+            [
+                (
+                    r"/app/static/(.*)",
+                    AppStaticFileHandler,
+                    {"path": "%s/" % self._tmpdir.name},
+                )
+            ]
+        )
+
+    def test_static_files_200(self):
+        responses = [
+            self.fetch(f"/app/static/{self._filename}"),
+            self.fetch(f"/app/static/{self._js_filename}"),
+            self.fetch(f"/app/static/{self._symlink_inside_directory}"),
```

Comment:
Do you mind commenting each of these fetches to explain what they're testing? (What's the difference between `self._filename` and `self._js_filename`? Why are those different tests?)